### ISSUE TITLE
Keep parent in `flatten-bom` configuration

### ISSUE THREAD

**Issue by vy** [apache/logging-parent#37]
`flatten-bom` configuration of `flatten-maven-plugin` doesn't keep the `parent`. This causes build failures for artifacts whose dependencies need to be resolved at runtime. Consider the following problem experienced in `log4j-transform-maven-plugin`:

1. `log4j-transform-parent` depends on `log4j-transform-bom`
2. `log4j-transform-bom` depends on `logging-parent`
3. `logging-parent` contains `dependencyManagement`, etc. that are used by `log4j-transform-maven-plugin`
4. Dependencies of `log4j-transform-maven-plugin` et al. is resolved *at runtime*
5. Though at runtime, the deployed `-bom` is used, which is flattened and hence doesn't have a parent!
6. Hence, at runtime, all `logging-parent` logic is lost

To avoid this, `flatten-bom` configuration should better keep the parent using `<parent>keep</parent>` in `pomElements` block.

**Comment by vy** [comment]
Fixed in 041768f401c840b77f75062c2e61c76f60388644.

**Comment by vy** [comment]
Reproduction of the issue:

```
$ git clone https://github.com/apache/logging-log4j-tools.git && cd $_
$ git checkout 18425d95a4d150de905929d55bf55545d70f3d91
$ ./mvnw -N log4j-changelog:export@export-changelog -X
...
[INFO] --- log4j-changelog:0.5.0:export (export-changelog) @ log4j-tools-bom ---
[WARNING] The POM for org.apache.logging.log4j:log4j-changelog:jar:0.5.0 is invalid, transitive dependencies (if any) will not be available: 2 problems were encountered while building the effective model for org.apache.logging.log4j:log4j-changelog:0.5.0
[ERROR] 'dependencies.dependency.version' for org.osgi:osgi.annotation:jar is missing. @ 
[ERROR] 'dependencies.dependency.version' for com.github.spotbugs:spotbugs-annotations:jar is missing. @
```
